### PR TITLE
feat: implemente HomeTask2

### PR DESCRIPTION
**Pull request by litwin90**
Link to task: 
https://epam.sharepoint.com/sites/EPAMNode.jsGlobalMentoringProgram/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FHomework%2FModule%202%2FHomework%202%2Epdf&parent=%2Fsites%2FEPAMNode%2EjsGlobalMentoringProgram%2FShared%20Documents%2FGeneral%2FHomework%2FModule%202&p=true&originalPath=aHR0cHM6Ly9lcGFtLnNoYXJlcG9pbnQuY29tLzpiOi9zL0VQQU1Ob2RlLmpzR2xvYmFsTWVudG9yaW5nUHJvZ3JhbS9FVFFmWUZzTkUwWlBzemZyUGx2RjY0SUIyNWtBV2NfaE02ZWpzZXVDX3pNaWVRP3J0aW1lPUJIMGV0NUZLMkVn

Implemented rest api for User model according to the task requirements.

TS is used
Babal is not used
Eslint is not used

### CODE REVIEW

File: src/routes/user/user.router.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    const user = users.get(userId);
+
+    if (user && !user.isDeleted) {
+        res.status(HTTP_CODES.OK).json(user);
+    } else {
+        res.status(HTTP_CODES.NOT_FOUND).send(`User with id: ${userId} not found`);
+    }
+});
+
+userRouter.post('/', validateUser, (req, res) => {
```
Comment (KonanMentor) [inline src/routes/user/user.router.ts:34]: I wouldn't expect to specify `isDeleted` field when creating a user
```sh
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaa","age":52}' 'http://localhost:3000/user'
{"validationErrors":[{"keyword":"required","dataPath":"","schemaPath":"#/definitions/UserToValidate/required","params":{"missingProperty":"isDeleted"},"message":"should have required property 'isDeleted'"}]}% 
```

Comment (KonanMentor) [reply src/routes/user/user.router.ts:34]: I expect to be able to create a user after deleting it
```sh
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}' 'http://localhost:3000/user'
{"id":"bb79a794-be20-4eec-b491-ba15fa436a5e","login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}
$ curl -X DELETE 'http://localhost:3000/user/bb79a794-be20-4eec-b491-ba15fa436a5e'
User with id: bb79a794-be20-4eec-b491-ba15fa436a5e is deleted
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}' 'http://localhost:3000/user'
User with login aaaaa already exists% 
```

File: src/routes/user/user.router.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        const newUser = updateUser(user, { isDeleted: true });
+        users.set(newUser.id, newUser);
+
+        res.status(HTTP_CODES.OK).send(`User with id: ${userId} is deleted`);
+    } else {
+        res.status(HTTP_CODES.NOT_FOUND).send(`User with id: ${userId} not found`);
+    }
+});
+
+userRouter.post('/auto-suggest', (req, res) => {
```
Comment (KonanMentor) [inline src/routes/user/user.router.ts:79]: I guess, we shouldn't suggest deleted users
```sh
$ curl -X POST -H 'content-type: application/json' -d '{"loginSubstring":"a","limit":2}' 'http://localhost:3000/user/auto-suggest'
[{"id":"bb79a794-be20-4eec-b491-ba15fa436a5e","login":"aaaaa","age":52,"isDeleted":true,"password":"2aAaaaaaaaaaaa"}]% 
```

Comment (litwin90) [reply src/routes/user/user.router.ts:79]: @KonanMentor fixed

File: src/routes/user/user.router.ts (hunk 1 above)
Comment (litwin90) [reply src/routes/user/user.router.ts:34]: @KonanMentor fixed

**Comment by litwin90** [comment]
@KonanMentor added eslint to project

**Pull request merged**